aptos-move/framework/aptos-stdlib/sources/cryptography/algebra_bls12381.move

```diff
@@ -1,11 +1,214 @@
 /// This module defines marker types, constants and test cases for working with BLS12-381 curves
 /// using generic API defined in `algebra.move`.
 ///
-/// Currently supported BLS12-381 structures include field `Fr`.
+/// See https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-11#name-bls-curves-for-the-128-bit-
+/// for the full specification of BLS12-381 curves.
+///
+/// Currently-supported BLS12-381 structures include `Fq12`, `Fr`, `G1Affine`, `G2Affine` and `Gt`,
+/// along with their widely-used serialization formats,
+/// the pairing between `G1Affine`, `G2Affine` and `Gt`,
+/// and the hash-to-curve operations for `G1Affine` and `G2Affine` defined in https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16.
 module aptos_std::algebra_bls12381 {
     //
-    // Marker types and their serialization schemes begin.
```

Comment:
A few things we should discuss here:

1. Remove `Affine` everywhere to have shorter names and because it's unnecessary to have in the name. We can describe that the serialized point format is "affine" in the comments.
2. There are many information-only structs. This is good cause it helps explain everything else, but it'll make finding the right struct types harder for devs.
   - Let's create two sections (using the formatting from `algebra.move`): one for actual used struct markers and one for unused ones.
   - In the unused section, let's comment out the actual struct definitions. This will make sure that autocomplete tools don't confuse our Move developers.
3. Consider renaming `GXAffineParent` to GXNonPrime` (for all X)
4. Consider compressing `Uncompressed` and `Compressed` to `Uncompr` and `Compr`, respectively.
5. Consider renaming all format markers to start as `FormatGX` (or `FmtGX`) rather than `GXFormat`. This will make it easier in autocomplete.

Reply:
For 3, I don't really like i.e. `G1NonPrime` as `G1` implies the element is in G1 (false if it is not in the prime order subgroup) and `NonPrime` implies the element is not in the prime order subgroup (false if it is in the prime order subgroup). According to wikipedia "overgroup" is sometimes used for the group of a subgroup.

Reply:
Maybe `G1Full`?

Reply:
Moved all information-only structures into module documentation. No need to decide G1Overgroup or G1Full right now.